tests/lib/schema_validation_test.py

```diff
@@ -231,6 +231,153 @@ def test_over_maximum_total_vfs_is_invalid(
+                            Ethernet.SRIOV.VFS.MAC: "12:34:56:78:90:ab",
+                            Ethernet.SRIOV.VFS.SPOOF_CHECK: True,
+                            Ethernet.SRIOV.VFS.TRUST: False,
+                            Ethernet.SRIOV.VFS.MIN_TX_RATE: 1000,
+                            Ethernet.SRIOV.VFS.MAX_TX_RATE: 2000,
+                        }
+                    ],
+                },
+            }
+        )
+        libnmstate.validator.validate(default_data)
+
+    def test_sriov_vf_vlan_config_is_valid(self, default_data):
+        vlans_constants = Ethernet.SRIOV.VFS.VLANS
+        default_data[Interface.KEY][0].update(
+            {
+                Interface.TYPE: InterfaceType.ETHERNET,
+                Ethernet.SRIOV_SUBTREE: {
+                    Ethernet.SRIOV.TOTAL_VFS: 1,
+                    Ethernet.SRIOV.VFS_SUBTREE: [
+                        {
+                            Ethernet.SRIOV.VFS.ID: 1,
+                            Ethernet.SRIOV.VFS.MAC: "12:34:56:78:90:ab",
+                            Ethernet.SRIOV.VFS.SPOOF_CHECK: True,
+                            Ethernet.SRIOV.VFS.TRUST: False,
+                            Ethernet.SRIOV.VFS.MIN_TX_RATE: 1000,
+                            Ethernet.SRIOV.VFS.MAX_TX_RATE: 2000,
+                            Ethernet.SRIOV.VFS.VLANS_SUBTREE: [
+                                {
+                                    Ethernet.SRIOV.VFS.VLANS.ID: 2,
+                                    Ethernet.SRIOV.VFS.VLANS.PRIORITY: 0,
+                                    vlans_constants.PROTOCOL: "802.1Q",
+                                },
+                                {
+                                    Ethernet.SRIOV.VFS.VLANS.ID: 4,
+                                    Ethernet.SRIOV.VFS.VLANS.PRIORITY: 2,
+                                    vlans_constants.PROTOCOL: "802.1ad",
+                                },
+                            ],
+                        }
+                    ],
+                },
+            }
+        )
+        libnmstate.validator.validate(default_data)
+
+    @pytest.mark.parametrize("invalid_values", [-50, -1])
+    def test_sriov_below_vf_vlanid_minimum_is_invalid(
+        self, default_data, invalid_values
+    ):
+        default_data[Interface.KEY][0].update(
+            {
+                Interface.TYPE: InterfaceType.ETHERNET,
+                Ethernet.SRIOV_SUBTREE: {
+                    Ethernet.SRIOV.TOTAL_VFS: 1,
+                    Ethernet.SRIOV.VFS_SUBTREE: [
+                        {
+                            Ethernet.SRIOV.VFS.ID: 1,
+                            Ethernet.SRIOV.VFS.VLANS_SUBTREE: [
+                                {Ethernet.SRIOV.VFS.VLANS.ID: invalid_values}
+                            ],
+                        }
+                    ],
+                },
+            }
+        )
+
+        with pytest.raises(js.ValidationError, match=str(invalid_values)):
```

Comment:
The string of error is not part of API of jsonschema module. There is not need to test on that.

Reply:
This test does not exist anymore. Thanks!